smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/ReactiveMessagingExtension.java

```diff
@@ -79,6 +88,18 @@ void afterDeploymentValidation(@Observes AfterDeploymentValidation done, BeanMan
         CompletableFuture<Void> future = mediatorManager.initializeAndRun();
         try {
             future.get();
+
+            // NOTE: We do not validate @Stream annotations added by portable extensions
+            Set<String> names = registry.getPublisherNames();
+            for (InjectionPoint ip : streamInjectionPoints) {
+                String name = StreamProducer.getStreamName(ip);
+                if (!names.contains(name)) {
+                    done.addDeploymentProblem(new DeploymentException("No stream found for name: " + name + ", injection point: " + ip));
```

Comment:
We should probably add a test for this...

Reply:
````
@return the stream name, mandatory, non null and non-blank. It must matches one of the available streams.
```

It should be enforced. (working on it)

Reply:
Yes. What I'm saying is that we throw `DeploymentException` if there is a `@Stream` with unknown stream name but we don't have a test for this right now. I can add the test today.